UnityProject/Assets/Scripts/Objects/Machines/Gibber.cs

```diff
@@ -0,0 +1,158 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using HealthV2;
+using Mirror;
+using NaughtyAttributes;
+using Systems.Electricity;
+using UnityEngine;
+
+namespace Objects.Machines
+{
+	public class Gibber : MonoBehaviour, ICheckedInteractable<HandApply>, ICheckedInteractable<MouseDrop>, IExaminable, IRightClickable
+	{
+		[SerializeField] private ObjectContainer storage;
+		[SerializeField] private UniversalObjectPhysics physics;
+		[SerializeField] private Integrity machineIntegrity;
+		[SerializeField] private float timeToGib = 12f;
+		[SerializeField] private int produceMultiplier = 4;
+		[SerializeField] private int damagePerFrame = 20;
+		[SerializeField] private GameObject defaultProduce;
+		[SerializeField] private SpriteHandler lights;
+		[SerializeField] private SpriteDataSO lightsoff;
+		[SerializeField] private SpriteDataSO lightson;
+
+		private bool isRunning = false;
+
+		private Dictionary<GameObject, int> gibbed = new Dictionary<GameObject, int>();
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			if (DefaultWillInteract.Default(interaction, side) == false) return false;
+			//For people attacking the object so they don't accidentally trip it.
+			return interaction.Intent != Intent.Harm;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			if (interaction.HandObject != null)
+			{
+				storage.StoreObject(interaction.HandObject);
+				return;
+			}
+
+			isRunning = !isRunning;
+			if (isRunning) StartCoroutine(GibbingTime());
+		}
+
+		public bool WillInteract(MouseDrop interaction, NetworkSide side)
+		{
+			// Only interact with this if it's anchored.
+			if (physics.isNotPushable == false) return false;
+			//Checks if the player is in reach + if they are able to interact with this type of object.
+			return DefaultWillInteract.Default(interaction, side);
+		}
+
+		public void ServerPerformInteraction(MouseDrop interaction)
+		{
+			storage.StoreObject(interaction.DroppedObject);
+		}
+
+		private IEnumerator GibbingTime()
+		{
+			Chat.AddLocalMsgToChat("The gibber violently shakes as it shreds everything inside of it.", gameObject);
+			lights.SetSpriteSO(lightson);
+			int time = 0;
+			while (isRunning)
+			{
+				if(timeToGib <= time) break;
+				time++;
+				yield return WaitFor.Seconds(0.6f);
+				yield return CheckContentAndHarm();
+			}
+
+			storage.RetrieveObjects();
+			foreach (var products in gibbed)
+			{
+				_ = Spawn.ServerPrefab(products.Key, gameObject.TileWorldPosition().To3(), gameObject.RegisterTile().Matrix.transform,
+					null, Mathf.Min(1, products.Value));
```

Comment:
Is this supposed to limit the product count to 1? I think you might've meant to use `Mathf.Max`.

Reply:
Good catch, Will touch up upon it.